File: CampaignManager/Data/Initiatives/Scene.cs
namespace CampaignManager.Data.Initiatives;

public class Scene : EntityBase
{
    public required string UserId { get; set; }

    public required string Name { get; set; }

    public required string Description { get; set; }

    public string? Image { get; set; }

    public int Width { get; set; }

    public int Height { get; set; }

    public int? SquareSize { get; set; }

    public int CombatantTurn { get; set; }

    public int Round { get; set; }

    public virtual ICollection<Combatant> Combatants { get; set; } = Array.Empty<Combatant>();
}

File: CampaignManager.Frontend/src/common/Scene.fs
module Scene

type CombatantType =
    | Enemy
    | Ally
    | Player

type GameState =
    | CharacterSetup
    | Active

type Combatant =
    { InitiativeModifier: int
      Name: string
      ImageUrl: string
      PlayerType: CombatantType
      LocationX: int
      LocationY: int
      Health: int
      MaxHealth: int
      ArmorClass: int
      IsTokenBeingDragged: bool
      PreviousLocationX: int
      PreviousLocationY: int
      RolledInitiative: int option }

type Scene =
    { Id: int
      Name: string
      Description: string
      BackgroundImage: string
      Width: int
      Height: int
      SquareSize: int
      CombatantTurn: int
      Round: int
      Combatants: Combatant List
      GameState: GameState
      ShowGrid: bool
      GridSize: int }
